XUtils.c

```diff
@@ -303,3 +304,12 @@ ssize_t xReadfileat(openat_arg_t dirfd, const char* pathname, void* buffer, size
 
    return readfd_internal(fd, buffer, count);
 }
+
+#ifndef HAVE_BUILTIN_CLZ
+unsigned int prevPowerOf2(unsigned int x) {
+   for (unsigned int shift = 1; shift < sizeof(x) * CHAR_BIT; shift <<= 1) {
+      x |= (x >> shift);
+   }
+   return (x - (x >> 1));
```

Comment:
Returns `1` for `x=1`, but rather should `0` IMHO.

Reply:
Perhaps I should write a comment line of what this does:
"Return the nearest power of two that is not greater than x."
`x == 1` returning 1 is expected.
This behavior is chosen to match [std::bit_floor](https://en.cppreference.com/w/cpp/numeric/bit_floor) in C++20 and perhaps other libraries that implement similar function.
If x is a power of two, the function returns x itself (not `(x/2)`).
`x==0` might be a special case (there's no power of two not greater than 0) and some implementations chose to result in undefined behavior, but I chose to return 0 which might be the most reasonable value to return.